producer/callback.go

```diff
@@ -467,20 +468,24 @@ func NfSubscriptionStatusNotifyProcedure(notificationData models.NotificationDat
 
 	// If nrf caching is enabled, go ahead and delete the entry from the cache.
 	// This will force the amf to do nf discovery and get the updated nf profile from the nrf.
-	if amf_context.AMF_Self().EnableNrfCaching {
-		ok := nrf_cache.RemoveNfProfileFromNrfCache(nfInstanceId)
-		logger.ProducerLog.Tracef("nfinstance %v deleted from cache: %v", nfInstanceId, ok)
-	}
-	if subscriptionId, ok := amf_context.AMF_Self().NfStatusSubscriptions.Load(nfInstanceId); ok {
-		logger.ConsumerLog.Debugf("SubscriptionId of nfInstance %v is %v",nfInstanceId,subscriptionId.(string))
-		problemDetails, err := consumer.SendRemoveSubscription(subscriptionId.(string))
-		if problemDetails != nil {
-			logger.ConsumerLog.Errorf("Remove NF Subscription Failed Problem[%+v]", problemDetails)
-		} else if err != nil {
-			logger.ConsumerLog.Errorf("Remove NF Subscription Error[%+v]", err)
+	if notificationData.Event == models.NotificationEventType_DEREGISTERED {
```

Comment:
why do we need this check. will the amf instance down reach here with some other event type ?